pkg/kubelet/cm/cpumanager/cpu_assignment.go

```diff
@@ -223,14 +223,21 @@ type cpuAccumulator struct {
 	result cpuset.CPUSet
 
 	numaOrSocketsFirst numaOrSocketsFirstFuncs
+
+	// 'options' is the StaticPolicyOptions. This is used to control the cpu sorting result.
+	// The sequence of returned CPU IDs depends on the policy.
+	// By default, cpus is sorted by sortAvailableCPUs()
+	// If staticPolicy is `SpreadPhysicalCPUsPreferredOption`, cpu is sorted by sortAvailableNonConnectedCPUs()
+	options StaticPolicyOptions
```

Comment:
this works, but to keep the spirit of the surrounding code I'd use a function pointer or a wrapper object similar to `numaOrSocketsFirst` so the rest of the code can just keep calling `a.sortAvailableCPUs` as usual, which is dispatched to either `a.sortAvailableNonConnectedCPUs()` or `a.sortAvailableCPUsPacked` (the existing function renamed)

Reply:
Got it. I will make the change.